src/ECER.Resources.Accounts/Communications/CommunicationRepository.cs

```diff
@@ -26,81 +26,53 @@ public CommunicationRepository(EcerContext context, IMapper mapper, IObjecStorag
-      communications = communications.WhereIn(communication => communication.c.StatusCode!.Value, statuses);
+      communications = communications.WhereIn(item => item.StatusCode!.Value, statuses);
     }
 
     // Filtering by ID
-    if (query.ById != null) communications = communications.Where(r => r.c.ecer_CommunicationId == Guid.Parse(query.ById));
-    else
+    if (query.ById != null) communications = communications.Where(item => item.ecer_CommunicationId == Guid.Parse(query.ById));
+
+    // otherwise if parent id provided returning just child communications based on parent id
+    else if (query.ByParentId != null)
     {
-      // returning just child communications based on parent id
-      if (query.ByParentId != null)
-      {
-        communications = communications.Where(r => r.c.ecer_ParentCommunicationid.Id == Guid.Parse(query.ByParentId));
-        UnreadMessagesCount = communications.Where(comms => comms.c.ecer_InitiatedFrom == ecer_InitiatedFrom.Registry && comms.c.ecer_Acknowledged != true).ToList().Count;
-      }
-      // otherwise returning parent communications
-      else
-      {
-        var communicationsList = communications.ToList();
-        UnreadMessagesCount = communications.Where(comms => comms.c.ecer_InitiatedFrom == ecer_InitiatedFrom.Registry && comms.c.ecer_Acknowledged != true).ToList().Count;
-        var latestMessageDateByParentId = communicationsList.GroupBy(ct => ct.c.ecer_ParentCommunicationid.Id) // Group by parent ID
-                                                     .Select(group => new
-                                                     {
-                                                       ParentId = group.Key,
-                                                       LatestMessageNotifiedDate = group.Max(ct => ct.c.ecer_DateNotified),
-                                                       AreAllRead = !group.Any(r => r.c.ecer_InitiatedFrom == ecer_InitiatedFrom.Registry && r.c.ecer_Acknowledged != true),
-                                                     });
-
-        communications = communications.Where(c => c.c.ecer_IsRoot == true);
-        context.LoadProperties(communications.Select(comms => comms.c), ecer_Communication.Fields.ecer_bcgov_documenturl_CommunicationId_ecer_communication);
-        PortalComms = mapper.Map<IEnumerable<Communication>>(communications.Select(comms => comms.c).ToList());
-
-        foreach (var comm in PortalComms)
-        {
-          var matchingComm = latestMessageDateByParentId.SingleOrDefault(ld => ld.ParentId == Guid.Parse(comm.Id!));
-          if (matchingComm != null)
-          {
-            comm.LatestMessageNotifiedOn = matchingComm.LatestMessageNotifiedDate;
-            comm.IsRead = matchingComm.AreAllRead;
-          }
-        }
-      }
+      communications = communications.Where(item => item.ecer_ParentCommunicationid.Id == Guid.Parse(query.ByParentId) && item.ecer_IsRoot != true);
```

Comment:
Should get rid of " && item.ecer_IsRoot != true" here, otherwise base parent message will not get displayed in message thread